src/Patient_Dashboard/PatientDashboard.js

```diff
@@ -1,60 +1,203 @@
 
 const PatientDashboard = () => {
   const { id } = useParams();
+  const navigate = useNavigate();
   const location = useLocation();
   const patientData = location.state ? location.state.patientData : null;
+  const [visits, setVisits] = useState([]);
+  const [selectedDocumentPreviews, setSelectedDocumentPreviews] = useState({});
+  const visitDate = location.state ? location.state.visitDate : null;
+  const [selectedDocument, setSelectedDocument] = useState(null);
+
+  useEffect(() => {
+    fetchVisits();
+  }, []);
+  const fetchVisits = async () => {
+    try {
+      const visitResponse = await fetch(
+        `/openmrs/ws/rest/v1/visit?patient=${id}`
+      );
+      const visitData = await visitResponse.json();
+      setVisits(visitData.results);
+    } catch (error) {
+      console.error("Error fetching visits:", error);
+    }
+  };
+  const formatDate = (inputDate) => {
+    const date = new Date(inputDate);
+    const day = date.getDate();
+    const month = date.toLocaleString("default", { month: "short" });
+    const year = date.getFullYear();
+    return `${day}-${month}-${year}`;
+  };
+  const handleCapture = async () => {
+    if (selectedDocument) {
+      const formData = new FormData();
+      formData.append('file', selectedDocument);
+      try {
+        const uploadResponse = await fetch(
+          '/openmrs/ws/rest/v1/bahmnicore/visitDocument/uploadDocument',
+          {
+            method: 'POST',
+            body: formData,
+          }
+        );
+        if (uploadResponse.ok) {
+          console.log('Document uploaded successfully');
+          setSelectedDocument(null); // Clear the selected document
+          fetchVisits(); // Refresh the visits after upload
+        } else {
+          console.error('Error uploading document');
+        }
+      } catch (error) {
+        console.error('Error uploading document:', error);
+      }
+    } else {
+      console.log('No document selected');
+    }
+  };
+  
+  const handleDocumentUpload = (event, visitId) => {
+    const file = event.target.files[0];
+    if (file) {
+      setSelectedDocument(file); // Set the selected document
+      const previewUrl = URL.createObjectURL(file);
+      setSelectedDocumentPreviews((prevPreviews) => ({
+        ...prevPreviews,
+        [visitId]: prevPreviews[visitId]
+          ? [...prevPreviews[visitId], previewUrl]
+          : [previewUrl],
+      }));
+      // Pass the visit object to the UploadDocumentPage
+      const selectedVisit = visits.find((visit) => visit.uuid === visitId);
+      // const visitDate = formatDate(selectedVisit.display); // Extract visitDate
+      navigate(`/upload/${visitId}`, {
+        state: {
+          patientData,
+          visit: selectedVisit, // Pass the visit object here
+          selectedDocumentPreview: previewUrl,
+          visitDate: visitDate, // Pass the visitDate here
+        },
+      });
+    }
+  };
   const getInitials = (name) => {
```

Comment:
Requires unit tests to validate different cases.

Reply:
Also, ideally there should be a Patient Wrapper class, that returns a Patient object (in JS) with all relevant data. Here it seems PatientDashboard class is handling file stuff and patient name stuff in one place.